### PR TITLE
Stream mongodb charts data

### PR DESCRIPTION
**Pull request by Elkrival**
This pr updates the charts query from toArray to stream

### CODE REVIEW

File: server/controllers/chartsDataController/index.js [hunk 1]
```diff
@@ -21,21 +22,34 @@ const show = async (req, res, next) => {
...
+    const chartProcessor = new BarChartDataProcessor(
+      chart,
+      chartService.generateStudyTargetTotals(filters.sites)
+    )
+    const dataStream = await ParticipantsModel.allForAssessment(
       appDb,
       chart,
       filtersService
     )
+    dataStream.on('data', (doc) => chartProcessor.processDocument(doc))
```
Comment (mikestone14) [inline server/controllers/chartsDataController/index.js]: can we keep all this logic in the service so we're not polluting the controller with it?

**Comment by mikestone14** [comment]
@Elkrival what ticket is this closing?

**Comment by Elkrival** [comment]
@mikestone14 pr #630  and #631  close out ticket [#617](https://github.com/AMP-SCZ/dpdash/issues/617)

File: server/controllers/chartsDataController/index.js (hunk 1 above)
Comment (Elkrival) [reply server/controllers/chartsDataController/index.js]: [403d065](https://github.com/AMP-SCZ/dpdash/pull/631/commits/403d065498c9a44a8c0963abfe236bc601a2615c)
👍

**Review by mikestone14 — APPROVED** [review]

**Pull request merged**